lib/modules/us_core_guidance/clinicalnotes_sequence.rb

```diff
@@ -51,175 +41,168 @@ class USCoreR4ClinicalNotesSequence < SequenceBase
 
-          self.document_attachments = ClinicalNoteAttachment.new(resource_class) if document_attachments.nil?
+        message = 'Could not find these '
 
-          document_references = fetch_all_bundled_resources(reply)
+        unless type_missing.empty?
+          message += "DocumentReference types #{type_missing.join(', ')}"
+          message += ' and ' unless category_missing.empty?
+        end
 
-          document_references&.each do |document|
-            document&.content&.select { |content| !document_attachments.attachment.key?(content&.attachment&.url) }&.each do |content|
-              document_attachments.attachment[content.attachment.url] = document.id
-            end
+        message += "DiagnosticReport categories #{category_missing.join(', ')}." unless category_missing.empty?
+
+        skip message
+      end
+
+      def reply_with_status_search(search_params, resource_class, all_status)
+        reply = get_resource_by_params(versioned_resource_class(resource_class), search_params)
+
+        # If server require status, server shall return 400
+        # https://www.hl7.org/fhir/us/core/general-guidance.html#search-for-servers-requiring-status
+        if reply.code == 400
+          begin
+            parsed_reply = JSON.parse(reply.body)
+            assert parsed_reply['resourceType'] == 'OperationOutcome', 'Server returned a status of 400 without an OperationOutcome.'
+          rescue JSON::ParserError
+            assert false, 'Server returned a status of 400 without an OperationOutcome.'
           end
 
-          true
+          warning do
+            assert @instance.server_capabilities&.search_documented?(resource_class),
+                   %(Server returned a status of 400 with an OperationOutcome, but the
+                      search interaction for this resource is not documented in the
+                      CapabilityStatement. If this response was due to the server
+                      requiring a status parameter, the server must document this
+                      requirement in its CapabilityStatement.)
+          end
+
+          params_with_status = search_param.merge('status': all_status)
+          reply = get_resource_by_params(versioned_resource_class(resource_class), params_with_status)
         end
 
-        skip "No #{resource_class} resources with type #{category_code} appear to be available. Please use patients with more information." unless resources_found
-      end
+        assert_response_ok(reply)
+        assert_bundle_response(reply)
 
-      def test_clinical_notes_diagnostic_report(category_code)
-        skip_if_known_not_supported(:DiagnosticReport, [:search])
+        fetch_all_bundled_resources(reply)
+      end
 
-        resource_class = 'DiagnosticReport'
-        patient_ids = @instance.patient_ids.split(',')
+      def check_document_reference_required_type(patient_id)
+        resource_class = :DocumentReference.to_s
```

Comment:
```suggestion
        resource_class = 'DocumentReference'
```
Or is this to work around a frozen string issue?